src/main/scala/cromwell/engine/db/DataAccess.scala

```diff
@@ -32,7 +32,8 @@ trait DataAccess {
 
   def getExecutionBackendInfo(workflowId: WorkflowId, call: Call): Future[CallBackendInfo]
 
-  def updateExecutionBackendInfo(workflowId: WorkflowId, call: Call, backendInfo: CallBackendInfo): Future[Unit]
+  // TODO: This would be a lot neater if it took a CallKey as a parameter.
```

Comment:
I set out to change this and found myself untangling lots of spaghetti. So I added this and focused on the important ticket changes.

Now that's done, I'll have another attempt at despaghettification.